tests/test_dan.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import unittest
+import torch
+
+from confopt.oneshot.archsampler import DARTSSampler
+from confopt.searchspace import SearchSpace, NASBench201SearchSpace, DARTSSearchSpace
+from confopt.searchspace.common.mixop import DynamicAttentionNetwork
+from confopt.train import SearchSpaceHandler
+
+DEVICE = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+
+
+class TestDynamicAttentionExploration(unittest.TestCase):
+    def test_nb201_init_auxiliary(self) -> None:
+        # NB201 does not have introduce any new parameters
+        searchspace = NASBench201SearchSpace()
+        dan = DynamicAttentionNetwork(C=16, num_ops=5, attention_weight=1).to(DEVICE)
+        self._test_with_optimizer(searchspace, dan)
+        self._test_vanilla_forward(searchspace, dan)
+
+    def test_darts_init_auxiliary(self) -> None:
+        searchspace = DARTSSearchSpace()
+        dan = DynamicAttentionNetwork(C=16, num_ops=8, attention_weight=1)
```

Comment:
to(DEVICE) missing?

Reply:
this is not required as it is going inside the searchspace handler, and when calling adapt_search_space that moves it onto the device